gluten-core/src/main/java/io/glutenproject/vectorized/GeneralOutIterator.java

```diff
@@ -25,8 +25,11 @@
 
 public abstract class GeneralOutIterator implements AutoCloseable, Serializable {
   protected final AtomicBoolean closed = new AtomicBoolean(false);
+  protected final long handle;
 
-  public GeneralOutIterator() {}
+  public GeneralOutIterator(long handle) {
+    this.handle = handle;
+  }
```

Comment:
Let's avoid adding native-related terms like `handle` to `gluten-core` module. The module doesn't see any native stuffs by design.

Reply:
I think the principle should be documented (gluten-core: non-native, gluten-data: native) in future. And sorry if that didn't look clear to you.

Reply:
Make sense for me.

Reply:
Will update later.

Reply:
Done.